### PR TITLE
Modify how pipelines are verified in the integration tests 

### PR DESCRIPTION
**Pull request by edaena**
Use the commit ID instead of the build count to verify if a pipeline was successful

### CODE REVIEW

File: tests/functions.sh [hunk 1]
```diff
@@ -301,6 +301,55 @@ function verify_pipeline_with_poll () {
...
+    source_version=$6
+    end=$((SECONDS+$poll_timeout))
+    loop_result="unknown"
+
+    echo "Attempting to verify that the pipeline build for $pipeline_name is successful..."
+    pipeline_result=$(az pipelines build definition show --name $pipeline_name --org $AZDO_ORG_URL --p $AZDO_PROJECT)
+    pipeline_id=$(tr '"\""' '"\\"' <<< "$pipeline_result" | jq .id)
+    echo "$pipeline_name has pipeline id of $pipeline_id"
+
+    build_run_exists=$(az pipelines build list --definition-ids $pipeline_id --org https://dev.azure.com/edaenasj --p spkintegrationtests | jq -r --arg source_version "$source_version" '.[].sourceVersion | select(. == $source_version) != null')
```
Comment (samiyaakhtar) [inline tests/functions.sh]: `$AZDO_ORG_URL` and `$AZDO_PROJECT`

File: tests/functions.sh [hunk 2]
```diff
@@ -301,6 +301,55 @@ function verify_pipeline_with_poll () {
...
+
+    if [ "$build_run_exists" != "true" ]; then
+        echo "Commit ID $source_version was nof found in pipeline run $pipeline_name."
```
Comment (samiyaakhtar) [inline tests/functions.sh]: not

**Review by samiyaakhtar — COMMENTED** [review]
LGTM. Is there a smoke test build against this?

File: tests/functions.sh (hunk 1 above)
Comment (edaena) [reply tests/functions.sh]: Ah! Good catch Samiya

**Review by samiyaakhtar — APPROVED** [review]

**Review by mtarng — APPROVED** [review]

**Pull request merged**